server/src/com/cloud/server/StatsCollector.java

```diff
@@ -412,6 +428,36 @@ protected void runInContext() {
         }
     }
 
+    class HostOutOfBandManagementStatsCollector extends ManagedContextRunnable {
+        @Override
+        protected void runInContext() {
+            try {
+                s_logger.debug("HostOutOfBandManagementStatsCollector is running...");
+                List<OutOfBandManagementVO> outOfBandManagementHosts = outOfBandManagementDao.findAllByManagementServer(ManagementServerNode.getManagementServerId());
+                if (outOfBandManagementHosts == null) {
+                    return;
+                }
+                for (OutOfBandManagement outOfBandManagementHost : outOfBandManagementHosts) {
+                    Host host = _hostDao.findById(outOfBandManagementHost.getHostId());
+                    if (host == null) {
+                        continue;
+                    }
+                    if (outOfBandManagementService.isOutOfBandManagementEnabled(host)) {
+                        outOfBandManagementService.submitBackgroundPowerSyncTask(host);
+                    } else {
+                        if (outOfBandManagementHost.getPowerState() != OutOfBandManagement.PowerState.Disabled) {
+                            if (outOfBandManagementService.transitionPowerStateToDisabled(Collections.singletonList(host))) {
```

Comment:
Collapse these two ``if`` blocks into a single ``else if`` on line 447 to improve readability.